avocado/core/future/settings.py

```diff
@@ -0,0 +1,516 @@
+     default value as argument. Developers only need to register the default
+     value once, here when calling this methods.
+
+  2. User/System configuration files (/etc/avocado or ~/.avocado/): This is
+     configured by the user, on a more "permanent way".
+
+  3. Command-line options parsed in runtime. This is configured by the user, on
+     a more "temporary way";
+
+ATTENTION: This is a future module, and will be moved out from this package
+soon.
+"""
+
+import configparser
+import glob
+import json
+import os
+
+from ..settings_dispatcher import SettingsDispatcher
+from ...utils import path
+
+
+class SettingsError(Exception):
+    """
+    Base settings error.
+    """
+
+
+class DuplicatedNamespace(SettingsError):
+    """
+    Raised when a namespace is already registered.
+    """
+
+
+class ConfigFileNotFound(SettingsError):
+    """
+    Error thrown when the main settings file could not be found.
+    """
+
+    def __init__(self, path_list):
+        super(ConfigFileNotFound, self).__init__()
+        self.path_list = path_list
+
+    def __str__(self):
+        return ("Could not find the avocado config file after looking in: %s" %
+                self.path_list)
+
+
+class ConfigOption:
+    def __init__(self, namespace, help_msg, key_type=str, default=None,
+                 parser=None, short_arg=None, long_arg=None,
+                 positional_arg=False, choices=None, nargs=None,
+                 metavar=None, required=None, action=None):
+        self.namespace = namespace
+        self.help_msg = help_msg
+        self.key_type = key_type
+        self.default = default
+        self.parser = parser
+        self.short_arg = short_arg
+        self.long_arg = long_arg
+        self.positional_arg = positional_arg
+        self.choices = choices
+        self.nargs = nargs
+        self._metavar = metavar
+        self.required = required
+        self._action = action
+        self._value = None
+
+    @property
+    def action(self):
+        if self.key_type is bool:
+            # action is automatic when using bool types
+            if self.default is False:
+                return 'store_true'
+            else:
+                return 'store_false'
+        return self._action
+
+    @property
+    def section(self):
+        return '.'.join(self.namespace.split('.')[:-1])
+
+    @property
+    def key(self):
+        return self.namespace.split('.')[-1]
+
+    @property
+    def value(self):
+        if self._value:
+            return self._value
+        return self.default
+
+    @property
+    def name_or_tags(self):
+        if self.positional_arg:
+            return self.key
+        tags = []
+        if self.short_arg:
+            tags.append(self.short_arg)
+        if self.long_arg:
+            tags.append(self.long_arg)
+        return tags
+
+    @property
+    def argparse_type(self):
+        # type is automatic when using lists Argparse do not support list
```

Comment:
I think you meant: " type is automatic when using lists because Argparse does not support lists". This is minor, though.

Reply:
Fixed in v2